### PR TITLE
Add support for providing a URL sanitizer

### PR DESCRIPTION
**Pull request by marcospassos**
## Summary
Add support for providing a URL sanitizer.

### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: test/tab.test.ts [hunk 1]
```diff
@@ -164,13 +186,35 @@ describe('A tab', () => {
...
+
+        const sanitizer: UrlSanitizer = jest.fn().mockReturnValue(new URL(sanitizedUrl));
+        const tab = new Tab(tabId, true, sanitizer);
+        const listener = jest.fn();
+
+        tab.addListener('urlChange', listener);
+
+        window.history.pushState({}, 'Other page', '/other-page');
+
+        expect(sanitizer).toHaveBeenCalledWith('http://localhost/other-page');
```
Comment (Fryuni) [inline test/tab.test.ts:200]: Is this actually passing? The code is not calling the function, I don't see where the sanitizer would be called from the event listener

Comment (Fryuni) [reply test/tab.test.ts:200]: Ok, found it

**Review by Fryuni — APPROVED** [review]

File: src/tab.ts [hunk 2]
```diff
@@ -110,7 +117,7 @@ export default class Tab {
             const currentUrl = window.location.href;
 
             if (url !== currentUrl) {
-                listener(normalizeUri(currentUrl));
+                listener(currentUrl);
```
[the remark this answers is not in the document]
Comment (marcospassos) [reply src/tab.ts:120]: It's, but in an upper level (see answer below)

File: test/tab.test.ts (hunk 1 above)
Comment (marcospassos) [reply test/tab.test.ts:200]: Yes, I just [moved it to a higher level](https://github.com/croct-tech/sdk-js/pull/102/files#diff-e175ebe06b3b78c717703f4bd512d3ac3db2feac141e75571ed6e3e41b6f1279R67)

**Pull request merged**